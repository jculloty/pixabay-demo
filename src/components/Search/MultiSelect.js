import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Label from './Label';

class MultiSelect extends PureComponent {
  static propTypes = {
    config: PropTypes.object.isRequired,
    value: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
  };

  state = {
    value: [],
  }

  static getDerivedStateFromProps(nextProps) {
    return {
      value: nextProps.value ? nextProps.value.split(',') : [],
    };
  }

  handleChange = (event) => {
    const { checked, name } = event.target;
    const list = this.state.value.filter((item) => item !== name);

    if (checked) {
      list.push(name);
    }

    this.props.handleChange(this.props.config.name, list.join(','));
  }

  render() {
    const { options: list, name } = this.props.config;
    const checked = this.props.value.split(',');

    const options = list.sort().map((item) => (
      <div className="form-check form-check-inline" key={item}>
        <Label name={item} />
        <input id={name} type="checkbox" checked={checked.includes(item)} name={item} onChange={this.handleChange} className="form-control" />
      </div>
    ));

    return (
      <div>
        {options}
      </div>
    );
  }
}

export default MultiSelect;
