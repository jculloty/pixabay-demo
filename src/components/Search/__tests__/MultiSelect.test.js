import React from 'react';
import renderer from 'react-test-renderer';

import MultiSelect from '../MultiSelect';

const buildComponent = function(props = {}) {
  props = {
    ...{
      config: {
        name: 'testing',
        options: ['one', 'two', 'three'],
        default: 'one',
      },
      value: 'two,three',
      handleChange: () => {},
    },
    ...props,
  };

  return (<MultiSelect {...props} />);
};

describe('Search: MultiSelect', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Write more tests', () => {
    expect(true).toBe(false);
  });
});
