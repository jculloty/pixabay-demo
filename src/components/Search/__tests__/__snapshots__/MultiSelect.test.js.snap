// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search: MultiSelect renders correctly 1`] = `
<div>
  <span>
    <label
      htmlFor="one"
    >
      One
    </label>
    <input
      checked={false}
      id="testing"
      name="one"
      onChange={[Function]}
      type="checkbox"
    />
  </span>
  <span>
    <label
      htmlFor="three"
    >
      Three
    </label>
    <input
      checked={true}
      id="testing"
      name="three"
      onChange={[Function]}
      type="checkbox"
    />
  </span>
  <span>
    <label
      htmlFor="two"
    >
      Two
    </label>
    <input
      checked={true}
      id="testing"
      name="two"
      onChange={[Function]}
      type="checkbox"
    />
  </span>
</div>
`;
